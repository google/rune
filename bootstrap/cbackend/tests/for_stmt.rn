//  Copyright 2024 Google LLC.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

println "Basic for loop:"
for i = 1, i <= 4, i = i + 1 {
    println i
}

/* TODO
println
println "Simple Range iterator:"
for i in range(10) {
  println i
}

println
println "Simple Range iterator starting at non-zero:"
for i in range(1, 10) {
  println i
}

println
println "Simple Range iterator starting at non-zero and non-default increment:"
for i in range(1, 10, 2) {
  println i
}

println
println "Iterate over arrays:"
for i in ["alpha", "beta", "gamma"] {
  println i
}
*/

