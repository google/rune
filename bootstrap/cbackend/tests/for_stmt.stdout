Basic for loop:
1
2
3
4
